File: data-structures-and-algorithms/lab 4/task2.py
# Daim Bin Khalid
# 251686775
# Lab 5

# Task 2
# Given an array of integers a and an integer n, returns the number of occurrences of n in a. Also compute time complexity
# Time Complexity: O(n)
def occurrences(a, n):
    counter = 0
    if len(a) == 0:
        return counter
    elif n == a[0]:
        counter += 1
        return counter + occurrences(a[1:], n)
    else:
        return counter + occurrences(a[1:], n)


print(occurrences([9, 8, 7, 9, 2, 9], 3))

File: intro-to-c++/lab4/task2.cpp
/* Lab 4
Name: Daim Bin Khalid
Roll no: 251686775*/

// Task 2
#include <iostream>

using namespace std;

// func to count ones in a binary conversion
void countOnes(int value)
{
    int binary_value, total_ones;

    // loop runs till value is reduced down to 0
    while (value > 0)
    {
        // get a binary value
        binary_value = value % 2;
        cout << binary_value << endl;

        // if value is 1 add to counter
        if (binary_value == 1)
        {
            total_ones += 1;
        }

        // reduce a number to check in total value
        value /= 2;
    }

    cout << "Total number of ones: " << total_ones << endl;
}

int main()
{
    int value;

    cout << "Enter a base 10 value: ";
    cin >> value;
    countOnes(value);

    return 0;
}
